### PR TITLE
Display SHA-256 instead of SHA-1 for user certificates

### PR DESCRIPTION
**Pull request by joelverhagen**
Address https://github.com/NuGet/Engineering/issues/2758.

In general we are avoiding the use of SHA-1 anyway as a cleanliness practice. SHA-256 or greater is the hash algorithm recommendation.

# Old

image

# New

image

### CODE REVIEW

File: src/Bootstrap/dist/css/bootstrap-theme.css [hunk 1]
```diff
@@ -324,6 +324,10 @@ img.reserved-indicator-icon {
 .breadcrumb-menu h1 {
   margin-top: 0px;
 }
+.page-account-settings .small-fingerprint,
```
Comment (dtivel) [inline src/Bootstrap/dist/css/bootstrap-theme.css]: Per your screenshot, this results in a red fingerprint.  Use of a different color is to grab attention, and I don't think it's needed here.  Besides, red is already used elsewhere to mean expired.

I think the fingerprint should use the default font color (e.g.:  black, in your screenshot).

File: src/NuGetGallery/Views/Shared/_AccountCertificates.cshtml [hunk 2]
```diff
@@ -109,6 +109,16 @@
                             </tr>
                         </tbody>
                     </table>
+
+                    <p>
+                        The SHA-256 fingerprint can be found by calculating the SHA-256 hash of the DER encoded
```
Comment (dtivel) [inline src/NuGetGallery/Views/Shared/_AccountCertificates.cshtml]: ...by calculating the **hex-encoded** SHA-256 hash of the DER-encoded...

File: src/Bootstrap/dist/css/bootstrap-theme.css (hunk 1 above)
Comment (joelverhagen) [reply src/Bootstrap/dist/css/bootstrap-theme.css]: Good point. Fixed.

File: src/NuGetGallery/Views/Shared/_AccountCertificates.cshtml (hunk 2 above)
Comment (joelverhagen) [reply src/NuGetGallery/Views/Shared/_AccountCertificates.cshtml]: Done.

File: src/NuGet.Services.Entities/Certificate.cs [hunk 3]
```diff
@@ -18,6 +18,7 @@ public class Certificate : IEntity
         /// <summary>
         /// Gets or sets the SHA-1 thumbprint of the certificate.
         /// </summary>
+        [Obsolete("This property should not be used since SHA-1 usage is avoided in NuGetGallery.")]
         public string Sha1Thumbprint { get; set; }
```
Comment (cristinamanum) [inline src/NuGet.Services.Entities/Certificate.cs]: Why not removing it?

File: src/NuGetGallery/Services/CertificateService.cs [hunk 4]
```diff
@@ -59,7 +59,9 @@ public async Task<Certificate> AddCertificateAsync(HttpPostedFileBase file)
 
                     certificate = new Certificate()
                     {
+#pragma warning disable CS0618 // Only set the SHA1 thumbprint, for backwards compatibility. Never read it.
                         Sha1Thumbprint = certificateFile.Sha1Thumbprint,
```
Comment (cristinamanum) [inline src/NuGetGallery/Services/CertificateService.cs]: If not used why not removing it? What are the backwards compat concerns?

Comment (joelverhagen) [reply src/NuGetGallery/Services/CertificateService.cs]: Rollback for the short term.

Spoke to @skofman1 about this. This is a scoped fix for SDL purposes. We can consider clean up later.

File: src/NuGet.Services.Entities/Certificate.cs (hunk 3 above)
Comment (joelverhagen) [reply src/NuGet.Services.Entities/Certificate.cs]: See https://github.com/NuGet/NuGetGallery/pull/7614#discussion_r333665880.

**Pull request merged**